src/schema/printer.rs

```diff
@@ -217,11 +258,86 @@ mod tests {
     let expected =
 "message schema {
   OPTIONAL group foo {
-    REQUIRED INT32 f1;
-    OPTIONAL BYTE_ARRAY f2;
+    REQUIRED INT32 f1 (INT_32);
+    OPTIONAL BYTE_ARRAY f2 (UTF8);
   }
-  REPEATED FIXED_LEN_BYTE_ARRAY f3;
+  REPEATED FIXED_LEN_BYTE_ARRAY (12) f3 (INTERVAL);
 }";
     assert_eq!(&mut s, expected);
   }
+
+  #[test]
+  fn test_print_and_parse_primitive() {
+    let message = Type::new_group_type(
+      "root", None, LogicalType::NONE, vec![
+        Rc::new(Type::new_group_type("a0", Some(Repetition::REQUIRED), LogicalType::NONE, vec![
+
+          Rc::new(Type::new_group_type("a1", Some(Repetition::OPTIONAL), LogicalType::LIST, vec![
+            Rc::new(Type::new_primitive_type("a2", Repetition::REPEATED,
+              PhysicalType::BYTE_ARRAY, LogicalType::UTF8, 0, 0, 0, None).unwrap())
+          ], None).unwrap()),
+
+          Rc::new(Type::new_group_type("b1", Some(Repetition::OPTIONAL), LogicalType::LIST, vec![
+            Rc::new(Type::new_group_type("b2", Some(Repetition::REPEATED), LogicalType::NONE, vec![
+              Rc::new(Type::new_primitive_type("b3", Repetition::OPTIONAL,
+                PhysicalType::INT32, LogicalType::NONE, 0, 0, 0, None).unwrap()),
+              Rc::new(Type::new_primitive_type("b4", Repetition::OPTIONAL,
+                PhysicalType::DOUBLE, LogicalType::NONE, 0, 0, 0, None).unwrap())
+            ], None).unwrap()),
+          ], None).unwrap())
+
+        ], None).unwrap()),
+      ], None
+    ).unwrap();
+
+    let mut s = String::new();
```

Comment:
Since this code is shared, maybe extract it into a separate function?

Reply:
Yes, will do. Thanks.